docs/static/transforming-data.asciidoc

```diff
@@ -0,0 +1,520 @@
+[[transformation]]
+== Transforming Data
+
+With over 200 plugins in the Logstash plugin ecosystem, it's sometimes
+challenging to choose the best plugin to meet your data processing needs.
+In this section, we've collected a list of popular plugins and organized them
+according to their processing capabilities:
+
+* <<core-operations>>
+* <<data-deserialization>>
+* <<field-extraction>>
+* <<lookup-enrichment>>
+
+[[core-operations]]
+=== Performing Core Operations
+
+The plugins described in this section are useful for core operations, such as
+aggregating, mutating, and dropping events.
+
+Also see <<filter-plugins>> for the full list of available filter plugins. 
+
+<<plugins-filters-aggregate,aggregate filter>>::
+
+Aggregates information from several events (typically log lines) that belong to
+the same task and pushes the aggregated information into a final task event.
+For example:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
```

Comment:
@jordansissel Right. I understand your concern. I've sorted the options alphabetically just because I think it's easier for people to scan. This topic isn't a reference topic, though, so I would be OK with listing the plugins in order of usefulness. Should I use the order that Alvin gives here? https://github.com/elastic/logstash/issues/6820